src/runtime_src/core/edge/user/aie/aie.cpp

```diff
@@ -89,25 +90,129 @@ Aie::Aie()
         }
     }
 
-#if 0
-    /* TODO Register Event, Error XRT handler here ? */
-    xaie_register_event_handler(event, module, xrtAieEventCallBack);
-    xaie_register_error_notification(error, module, xrtAieErrorCallBack, arg);
-#endif
+    /* Disable AIE interrupts */
+    u32 reg = XAieGbl_NPIRead32(XAIE_NPI_ISR);
+    XAieGbl_NPIWrite32(XAIE_NPI_ISR, reg);
+
+    if (XAieTile_EventsHandlingInitialize(&aieInst) != XAIE_SUCCESS)
+        throw xrt_core::error("Failed to initialize AIE Events Handling.");
```

Comment:
Is there a particular error code that should end up getting propagated out of the C-API itself?  If so, then add that error code to the exception.

Reply:
The AIE driver just returns XAIE_FAILURE (which is 1). And reading its code, it is not clear to me what error code should be put here.